tests/integration/helpers/cluster.py

```diff
@@ -498,190 +813,318 @@ def copy_file_to_container(self, container_id, local_path, dest_path):
-            self.hdfs_api = HDFSApi(user="root",
-                               timeout=timeout,
-                               kerberized=True,
-                               principal="root@TEST.CLICKHOUSE.TECH",
-                               keytab=keytab,
-                               krb_conf=krb_conf,
-                               host="kerberizedhdfs1",
-                               protocol="http",
-                               proxy_port=50070,
-                               data_port=1006,
-                               hdfs_ip=hdfs_ip,
-                               kdc_ip=kdc_ip)
+            # logging.debug("kdc_ip ", kdc_ip)
+            hdfs_api = HDFSApi(user="root",
+                              timeout=timeout,
+                              kerberized=True,
+                              principal="root@TEST.CLICKHOUSE.TECH",
+                              keytab=keytab,
+                              krb_conf=krb_conf,
+                              host="localhost",
+                              protocol="http",
+                              proxy_port=self.hdfs_kerberized_name_port,
+                              data_port=self.hdfs_kerberized_data_port,
+                              hdfs_ip=hdfs_ip,
+                              kdc_ip=kdc_ip)                         
         else:
-            self.hdfs_api = HDFSApi(user="root", host="hdfs1")
+            logging.debug("Create HDFSApi host={}".format("localhost"))
+            hdfs_api = HDFSApi(user="root", host="localhost", data_port=self.hdfs_data_port, proxy_port=self.hdfs_name_port)
+        return hdfs_api
+
+    def wait_kafka_is_available(self, kafka_docker_id, kafka_port, max_retries=50):
+        retries = 0
+        while True:
+            if check_kafka_is_available(kafka_docker_id, kafka_port):
+                break
+            else:
+                retries += 1
+                if retries > max_retries:
+                    raise Exception("Kafka is not available")
+                logging.debug("Waiting for Kafka to start up")
+                time.sleep(1)
 
 
-    def wait_hdfs_to_start(self, timeout=60):
+    def wait_hdfs_to_start(self, hdfs_api, timeout=300):
         start = time.time()
         while time.time() - start < timeout:
             try:
-                self.hdfs_api.write_data("/somefilewithrandomname222", "1")
-                print("Connected to HDFS and SafeMode disabled! ")
+                hdfs_api.write_data("/somefilewithrandomname222", "1")
+                logging.debug("Connected to HDFS and SafeMode disabled! ")
                 return
             except Exception as ex:
-                print("Can't connect to HDFS " + str(ex))
+                logging.debug("Can't connect to HDFS " + str(ex))
                 time.sleep(1)
 
         raise Exception("Can't wait HDFS to start")
 
-    def wait_mongo_to_start(self, timeout=30):
+    def wait_mongo_to_start(self, timeout=180):
```

Comment:
I think we have to move such staff into a separate file. But only in the following PR.

Reply:
+1 
I tried to move all services setup and wait_to_start from main logic.
cluster.py is far from cluster definition :)